core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/messaginglayer/FlowControl.scala

```diff
@@ -73,11 +74,15 @@ class FlowControl {
             .getOrElseUpdate(receiverId, new mutable.Queue[WorkflowMessage]())
             .isEmpty
         ) {
-          receiverIdToCredits(receiverId) = receiverIdToCredits(receiverId) - 1
+          receiverIdToCredits(receiverId) = receiverIdToCredits(receiverId) - getInMemSize(
+            msg.asInstanceOf[WorkflowDataMessage]
+          ).intValue()
```

Comment:
we used to define credit as `number of tuples`. now the credit is defined as `in memory size`. Please make sure to update method names and comments more meaningful to avoid confusion.

Here you are using credit - in memory size which may confuse readers.